lib/client.js

```diff
@@ -572,11 +572,14 @@ class Client {
   search({ resourceType, compartment, searchParams, headers } = {}) {
     if (compartment && resourceType) {
       return this.compartmentSearch({ resourceType, compartment, searchParams, headers });
-    } else if (resourceType && searchParams) {
-      return this.resourceSearch({ resourceType, searchParams, headers });
+    } else if (resourceType) {
```

Comment:
This doesn't seem right. I think `searchParams` are only optional for a compartment search.

Reply:
I guess I'm confused then. What would I do if I wanted to use fhir-kit-client to hit `/Patient` with no query params? Currently you're required to pass an empty `searchParams` which seems like an odd design decision. (Or should I be using some other method?)

Reply:
Is that a valid fhir request?

Reply:
It's worked on every FHIR dev server that I've played with. I'm not sure how else you would semantically ask for "Give me every Patient that you've got/I'm allowed to access"

Reply:
Also https://www.hl7.org/fhir/http.html#search makes it look like the parameters are optional